doc/examples/multi_app_repo.rst

```diff
@@ -0,0 +1,178 @@
+    include($ENV{XMOS_CMAKE_PATH}/xcommon.cmake)
+    project(sw_multiapp)
+
+    add_subdirectory(app_multiapp0)
+    add_subdirectory(app_multiapp1)
+
+`sandbox/sw_multiapp/deps.cmake`
+
+.. code-block:: cmake
+
+    set(APP_DEPENDENT_MODULES "lib_mod0"
+                              "lib_mod1"
+                              "lib_mod2")
+
+`sandbox/sw_multiapp/app_multiapp0/CMakeLists.txt`
+
+.. code-block:: cmake
+
+    cmake_minimum_required(VERSION 3.21)
+    include($ENV{XMOS_CMAKE_PATH}/xcommon.cmake)
+    project(multiapp0)
+
+    set(APP_HW_TARGET XCORE-AI-EXPLORER)
+    include(${CMAKE_CURRENT_LIST_DIR}/../deps.cmake)
+    set(XMOS_SANDBOX_DIR ${CMAKE_CURRENT_LIST_DIR}/../..)
+
+    XMOS_REGISTER_APP()
+
+`sandbox/sw_multiapp/app_multiapp0/src/main.c`
+
+.. code-block:: c
+
+    #include "mod0.h"
+    #include "mod1.h"
+
+    int main() {
+        mod0();
+        mod1();
+        return 0;
+    }
+
+`sandbox/sw_multiapp/app_multiapp1/CMakeLists.txt`
+
+.. code-block:: cmake
+
+    cmake_minimum_required(VERSION 3.21)
+    include($ENV{XMOS_CMAKE_PATH}/xcommon.cmake)
+    project(multiapp1)
+
+    set(APP_HW_TARGET XCORE-AI-EXPLORER)
+    include(${CMAKE_CURRENT_LIST_DIR}/../deps.cmake)
+    set(XMOS_SANDBOX_DIR ${CMAKE_CURRENT_LIST_DIR}/../..)
+
+    XMOS_REGISTER_APP()
+
+`sandbox/sw_multiapp/app_multiapp1/src/main.c`
+
+.. code-block:: c
+
+    #include "mod0.h"
+    #include "mod2.h"
+
+    int main() {
+        mod0();
+        mod2();
+        return 0;
+    }
+
+`sandbox/lib_mod0/lib_mod0/lib_build_info.cmake`
+
+.. code-block:: cmake
+
+    set(LIB_NAME lib_mod0)
+    set(LIB_VERSION 1.0.0)
+    set(LIB_INCLUDES api)
+    set(LIB_DEPENDENT_MODULES "")
+
+    XMOS_REGISTER_MODULE()
+
+`sandbox/lib_mod1/lib_mod1/lib_build_info.cmake`
+
+.. code-block:: cmake
+
+    set(LIB_NAME lib_mod1)
+    set(LIB_VERSION 1.0.0)
+    set(LIB_INCLUDES api)
+    set(LIB_DEPENDENT_MODULES "")
+
+    XMOS_REGISTER_MODULE()
+
+`sandbox/lib_mod2/lib_mod2/lib_build_info.cmake`
+
+.. code-block:: cmake
+
+    set(LIB_NAME lib_mod2)
+    set(LIB_VERSION 1.0.0)
+    set(LIB_INCLUDES api)
+    set(LIB_DEPENDENT_MODULES "")
+
+    XMOS_REGISTER_MODULE()
+
+
+Build instructions
+""""""""""""""""""
+
+Commands to build and run both applications, from working directory ``sandbox/sw_multiapp``:
+
+.. code-block:: console
+
+    cmake -G "Unix Makefiles" -B build
+    cd build
+    xmake
+
+The build products are ``app_multiapp0/bin/multiapp0.xe`` and ``app_multiapp1/bin/multiapp1.xe``.
+
+Alternatively, a single application can be configured and built. From working directory
+``sandbox/sw_multiapp/app_multiapp1``:
+
+.. code-block:: console
+
+    cmake -G "Unix Makefiles" -B build
+    cd build
+    xmake
+
+The build product is ``bin/multiapp1.xe``. Application ``app_multiapp0`` has not been built.
+
+Best practice
+"""""""""""""
+
+The union of the dependent modules lists for all applications in the repository should be listed
```

Comment:
Might be best to describe the problem this section is trying to address before describing the solution?

Reply:
Done.